api/config/v1/config.go

```diff
@@ -52,6 +53,15 @@ func NewConfig(c *cli.Context, flags []cli.Flag) (*Config, error) {
 	}
 
 	config.Flags.UpdateFromCLIFlags(c, flags)
+	// TODO: This is currently not at the flags level?
+	// Does this mean that we should move UpdateFromCLIFlags to function off Config?
+	if c.IsSet("imex-default-channel-strategy") {
```

Comment:
Would it simplify things to have the config option be a list of the IMEX channels to inject (if they exist)?

Reply:
I don't think this is a strong enough signal if we want to have an "auto" mode. We could get around this by allowing users to specify another flag (`imex.must-inject` or `imex.required`) that controls this behaviour. 

Note that the `TODO` here is more about the config API than IMEX in general and the same questions / comment would be applicable even if we do use something like `imex.defaultChannelIndex`.

Reply:
I guess I just don't like the "switch" of modes. I'm also not a huge fan of them in the toolkit with the "auto" vs. "legacy" vs. "cdi" modes, but that ship has sailed. I would like to avoid it here if we can, and introduce as simple an API as possible without adding too much complexity in the code.

Reply:
The reason for the complexity is that this is expected to work "everywhere" and we are shifting the responsibility for making a decision into the plugin instead of requiring that the GPU Operator (or the user) make a configuration decision. This was also the case in the toolkit. We didn't want to have to tell users on non-NVML platforms that they need to make changes and since the support matrix is complex, the implementation is more complex than what it needs to be.

Regardless, I'll spend some time thinking about this tomorrow. The injection semantics as covered in this PR are simple enough and as such the discussion is more about the API.